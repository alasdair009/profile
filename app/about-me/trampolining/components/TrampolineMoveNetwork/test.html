<html lang="en">
  <head>
    <style>
      .videos {
        display: flex;
        height: 100vh;
        width: 100vw;
      }
      .videos div {
        align-items: center;
        display: flex;
        flex-direction: column;
        justify-content: center;
        width: 50%;
      }
      iframe {
        aspect-ratio: 16/9;
        width: 90%;
      }
    </style>
  </head>
  <body>
    <div class="videos">
      <div>
        <iframe
          id="standard"
          src=""
          data-src="https://www.youtube.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Standard Embed"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goStandard">Standard</button>
      </div>
      <div>
        <iframe
          id="privacy"
          src=""
          data-src="https://www.youtube-nocookie.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Enhanced privacy"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goPrivacy">Privacy</button>
      </div>
    </div>
    <script>
      const goStandard = document.getElementById("goStandard");
      const standardIFrame = document.getElementById("standard");
      const goPrivacy = document.getElementById("goPrivacy");
      const privacyIFrame = document.getElementById("privacy");
      goStandard.addEventListener(
        "click",
        function () {
          standardIFrame.setAttribute("src", standardIFrame.getAttribute("data-src"));
        },
        false
      );
      goPrivacy.addEventListener(
        "click",
        function () {
          privacyIFrame.setAttribute("src", privacyIFrame.getAttribute("data-src"));
        },
        false
      );
    </script>
  </body>
</html>
