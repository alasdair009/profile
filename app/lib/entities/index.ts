"use client";

import styled from "styled-components";

export * from "@/entities";
export { Splash } from "./Splash";

export const FireWrapper = styled.div`
  display: flex;
  flex-direction: column;
  height: 100%;
  position: absolute;
  width: 100%;
`;
export const FirePadding = styled.div<{ $backgroundColor: string }>`
  background: ${({ $backgroundColor }) => $backgroundColor};
  flex: 1;
`;
