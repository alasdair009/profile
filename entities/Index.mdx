import { Meta } from "@storybook/blocks";
import Image from "next/image";
import amLogo from "./assets/am.svg";
import { rem } from "polished";

<Meta title="Alasdair Macrae Design" />

# Alasdair Macrae Design

<Image
  src={amLogo}
  alt="Green and white AM letters"
  width={0}
  height={0}
  style={{ maxWidth: rem(240), width: "100%", height: "auto" }}
/>

I am a keen supported of Storybook js using it across the majority of projects I build from websites, web-apps and even desktop applications.
