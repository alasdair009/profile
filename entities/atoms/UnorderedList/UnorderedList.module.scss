@use "../../../app/styles/colors";
@use "../../../app/styles/dimensions";
@use "../../../app/styles/typography";
@use "../../../app/styles/helpers/rem";

.root {
  display: flex;
  flex-direction: column;
  gap: rem.toRem(map-get(dimensions.$sizes, s8));
  list-style: none;
  margin: 0 auto;
  max-width: rem.toRem(dimensions.$globalTextMaxWidth);
  text-align: var(--align);
  width: 100%;

  li {
    font-size: rem.toRem(map-get(typography.$fontSizes, medium));
    padding-left: rem.toRem(map-get(dimensions.$sizes, s16));
    position: relative;
    width: 100%;

    &::before {
      background: colors.$colorWhiteGhost;
      content: "";
      height: calc(100% - rem.toRem(map-get(typography.$fontSizes, medium)));
      left: 0;
      min-height: rem.toRem(map-get(typography.$fontSizes, medium));
      position: absolute;
      top: 50%;
      transform: translateY(-50%);
      width: rem.toRem(map-get(dimensions.$sizes, s2));
    }
  }
}
