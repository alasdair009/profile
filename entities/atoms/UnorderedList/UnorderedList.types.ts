import { HTMLAttributes, ReactNode } from "react";

export type UnorderedListProps = {
  align?: "left" | "right" | "center";
  children: ReactNode[];
} & HTMLAttributes<HTMLUListElement>;
