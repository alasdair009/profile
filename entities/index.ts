export { borderRadii } from "./design-tokens/effects";
export { colors } from "./design-tokens/colors";
export { fonts } from "./design-tokens/typography";
export { sizes } from "./design-tokens/dimensions";

export { Button } from "./atoms/Button";
export { Fire } from "./atoms/Fire";
export { Heading } from "./atoms/Heading";
export { Link } from "./atoms/Link";
export { Orb } from "./atoms/Orb";
export { Paragraph } from "./atoms/Paragraph";
export { Spacer } from "./atoms/Spacer";
export { TextMask } from "./atoms/TextMask";

export { Container } from "./organisms/Container";
export { ContentPlate } from "./organisms/ContentPlate";
export { Header } from "./organisms/Header";
export { Splash } from "./organisms/Splash";
