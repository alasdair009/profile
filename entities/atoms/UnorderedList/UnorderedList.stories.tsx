import { UnorderedList } from "./UnorderedList";
import { Meta, StoryObj } from "@storybook/react";
import { expect, within } from "@storybook/test";
import { TextMask } from "@/entities";

const meta: Meta<typeof UnorderedList> = {
  component: UnorderedList,
  args: {
    children: (
      <>
        <li>First item</li>
        <li>Second item</li>
        <li>Third item</li>
      </>
    ),
    align: "left",
  },
  argTypes: {
    children: {
      type: "string",
    },
  },
};
export default meta;

export const Default: StoryObj<typeof UnorderedList> = {
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    const listElement = canvas.getByTestId(UnorderedList.name);

    await expect(listElement).toBeInTheDocument();
    await expect(listElement.children).toHaveLength(3);
    await expect(listElement.children[0]).toHaveTextContent("First item");
    await expect(listElement.children[1]).toHaveTextContent("Second item");
    await expect(listElement.children[2]).toHaveTextContent("Third item");
  },
};
