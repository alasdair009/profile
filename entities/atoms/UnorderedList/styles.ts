"use client";

import styled from "styled-components";
import { colors, fontSizes, globalTextMaxWidth, sizes } from "@/entities";
import { rem } from "polished";
import { UnorderedListProps } from "@/entities/atoms/UnorderedList/UnorderedList.types";

export const Root = styled.ul<{ $align: UnorderedListProps["align"] }>`
  display: flex;
  flex-direction: column;
  gap: ${sizes.s8.rem};
  list-style: none;
  margin: 0 auto;
  max-width: ${rem(globalTextMaxWidth)};
  text-align: ${({ $align }) => $align};
  width: 100%;

  li {
    font-size: ${fontSizes.medium.rem};
    padding-left: ${sizes.s16.rem};
    position: relative;
    width: 100%;

    &::before {
      background: ${colors.whiteGhost};
      content: "";
      height: calc(100% - ${fontSizes.medium.rem});
      left: 0;
      min-height: ${fontSizes.medium.rem};
      position: absolute;
      top: 50%;
      transform: translateY(-50%);
      width: ${sizes.s2.rem};
    }
  }
`;
