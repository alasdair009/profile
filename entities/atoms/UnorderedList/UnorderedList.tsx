import { Root } from "./styles";
import { UnorderedListProps } from "@/entities/atoms/UnorderedList/UnorderedList.types";

/**
 * HTML unordered list.
 */
export function UnorderedList({
  align = "left",
  children,
  ...rest
}: UnorderedListProps) {
  return (
    <Root $align={align} {...rest}>
      {children}
    </Root>
  );
}
