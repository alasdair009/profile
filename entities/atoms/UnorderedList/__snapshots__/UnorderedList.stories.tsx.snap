// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/UnorderedList Default play-test 1`] = `
<ul data-testid="UnorderedList"
    class="sc-hDcvty eHmOfp"
>
  <li>
    First item
  </li>
  <li>
    Second item
  </li>
  <li>
    Third item
  </li>
</ul>
`;
